### DISCUSSION TITLE
Can you give me a example?

### DISCUSSION THREAD

**Discussion by luckyhorse1** [MikePopoloski/slang#379] [Q&A] [answered]
- Lookup.cpp
```c++
// This is a hierarchical lookup unless it's the first component in the path and the
// current scope is either an interface port or a package
result.isHierarchical = symbol->kind != SymbolKind::InterfacePort &&
                                  symbol->kind != SymbolKind::Package &&
                                  symbol->kind != SymbolKind::CompilationUnit;
```
I cannot find a example for this.   
Is this code consistent with the comment?

And whether "f.x" is hierarchical below?
```c++
module m();
	function f;
		int x;
	endfunction
	int a = f.x;
endmodule
```

**Comment by MikePopoloski** [comment, marked as answer]
'f' in your example is a subroutine, so it is hierarchical. The code + comment seem clear to me -- it's hierarchical unless it's an interface port, a package, or a compilation unit.

**Reply by luckyhorse1** [reply]
so,“ an interface port, a package, or a compilation unit ” ，can you give me their example in hierarchical form, thank you

**Comment by luckyhorse1** [comment]
Is m.a hierarchical ?
```
module m;
  int a;
  int b = m.a;
endmodule
```

**Reply by MikePopoloski** [reply]
Is 'm' an interface port, a package, or a compilation unit?

**Reply by luckyhorse1** [reply]
m is a module instance.

**Reply by MikePopoloski** [reply]
Right, so it is hierarchical based on the condition above.

**Reply by luckyhorse1** [reply]
```
if (symbol->kind == SymbolKind::InstanceArray || symbol->kind == SymbolKind::Instance) {
      result.isHierarchical = symbol->getParentScope() != &context.scope;
  }
```

**Reply by MikePopoloski** [reply]
Yeah, there is a special case for accessing members in the same scope. It seems like you're well equipped to answer these questions yourself.

**Reply by luckyhorse1** [reply]
ok, but why interface port, a package, or a compilation unit be excluded, can you show some example?

**Reply by MikePopoloski** [reply]
This is just based on language rules. If you're looking for examples of things like package accesses, you can look up more general documentation or tutorials on SystemVerilog.

**Reply by luckyhorse1** [reply]
ok, thank you
